backend-rust/src/graphql_api.rs

```diff
@@ -1834,14 +1835,112 @@ impl SearchResult {
 
     async fn accounts(
         &self,
-        #[graphql(desc = "Returns the first _n_ elements from the list.")] _first: Option<i32>,
+        ctx: &Context<'_>,
+        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: Option<u64>,
         #[graphql(desc = "Returns the elements in the list that come after the specified cursor.")]
-        _after: Option<String>,
-        #[graphql(desc = "Returns the last _n_ elements from the list.")] _last: Option<i32>,
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
         #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
-        _before: Option<String>,
+        before: Option<String>,
     ) -> ApiResult<connection::Connection<String, Account>> {
-        todo_api!()
+        let account_address_regex: Regex = Regex::new(r"^[1-9A-HJ-NP-Za-km-z]{1,50}$").unwrap();
+        let pool = get_pool(ctx)?;
+        let query = ConnectionQuery::<i64>::new(first, after, last, before, 10)?;
+        let mut connection = connection::Connection::new(false, false);
+        if !account_address_regex.is_match(&self.query) {
+            return Ok(connection);
+        }
+
+        let results = if let Ok(parsed_address) =
+            concordium_rust_sdk::common::types::AccountAddress::from_str(&self.query)
+        {
+            if let Some(account) = sqlx::query_as!(
+                Account,
+                "SELECT
+                index,
+                transaction_index,
+                address,
+                amount,
+                delegated_stake,
+                num_txs,
+                delegated_restake_earnings,
+                delegated_target_baker_id
+            FROM accounts
+            WHERE
+                address = $1",
+                parsed_address.to_string()
+            )
+            .fetch_optional(pool)
+            .await?
+            {
+                connection.edges.push(connection::Edge::new(account.index.to_string(), account));
+            }
+            return Ok(connection);
+        };
+        let accounts = sqlx::query_as!(
+            Account,
+            r#"
+                SELECT * FROM (SELECT
+                    index,
+                    transaction_index,
+                    address,
+                    amount,
+                    delegated_stake,
+                    num_txs,
+                    delegated_restake_earnings,
+                    delegated_target_baker_id
+                FROM accounts
+                WHERE
+                    address LIKE $5 || '%'
```

Comment:
I am looking into https://www.postgresql.org/docs/current/pgtrgm.html to optimise this